sqrl-testing/sqrl-integration-tests/src/test/resources/usecases/avro-schema/avro-schema/schema.avsc

```diff
@@ -0,0 +1,370 @@
+//      "name": "nullableTimeMillisField",
+//      "type": ["null", {
+//        "type": "int",
+//        "logicalType": "time-millis"
+//      }],
+//      "default": null,
+//      "doc": "A nullable time field with millisecond precision"
+//    },
+//    {
+//      "name": "timeMicrosField",
+//      "type": {
+//        "type": "long",
+//        "logicalType": "time-micros"
+//      },
+//      "doc": "A time field with microsecond precision"
+//    },
+//    {
+//      "name": "nullableTimeMicrosField",
+//      "type": ["null", {
+//        "type": "long",
+//        "logicalType": "time-micros"
+//      }],
+//      "default": null,
+//      "doc": "A nullable time field with microsecond precision"
+//    },
+//    {
+//      "name": "complexArrayField",
+//      "type": {
+//        "type": "array",
+//        "items": {
+//          "type": "record",
+//          "name": "ArrayItemRecord",
+//          "fields": [
+//            {
+//              "name": "itemFieldOne",
+//              "type": "int"
+//            },
+//            {
+//              "name": "itemFieldTwo",
+//              "type": "string"
+//            }
+//          ]
+//        }
+//      },
+//      "doc": "An array of complex nested records"
+//    },
+//    {
+//      "name": "nullableComplexArrayField",
+//      "type": ["null", {
+//        "type": "array",
+//        "items": {
+//          "type": "record",
+//          "name": "NullableArrayItemRecord",
+//          "fields": [
+//            {
+//              "name": "nullableItemFieldOne",
+//              "type": ["null", "double"],
+//              "default": null
+//            },
+//            {
+//              "name": "nullableItemFieldTwo",
+//              "type": ["null", "boolean"],
+//              "default": null
+//            }
+//          ]
+//        }
+//      }],
+//      "default": null,
+//      "doc": "A nullable array of complex nested records"
+//    },
+//    {
+//      "name": "multiNestedRecord",
+//      "type": {
+//        "type": "record",
+//        "name": "MultiNestedRecord",
+//        "fields": [
+//          {
+//            "name": "nestedLevelOne",
+//            "type": {
+//              "type": "record",
+//              "name": "NestedLevelOne",
+//              "fields": [
+//                {
+//                  "name": "levelOneField",
+//                  "type": "string"
+//                },
+//                {
+//                  "name": "nestedLevelTwo",
+//                  "type": {
+//                    "type": "record",
+//                    "name": "NestedLevelTwo",
+//                    "fields": [
+//                      {
+//                        "name": "levelTwoField",
+//                        "type": "int"
+//                      }
+//                    ]
+//                  }
+//                }
+//              ]
+//            }
+//          }
+//        ]
+//      },
+//      "doc": "A multi-level nested record"
```

Comment:
Error in assuming all relation types have an array depth > 0 in flexible schema processing